src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -166,15 +167,42 @@ public void EmitDebugLoc(int nativeOffset, int fileId, int linueNumber, int colN
         }
 
         [DllImport(NativeObjectWriterFileName)]
-        private static extern void FlushDebugLocs(IntPtr objWriter, string methodName, int methodSize);
-        public void FlushDebugLocs(int methodSize)
+        private static extern void EmitDebugVar(IntPtr objWriter, string name, UInt32 typeIndex, bool isParam, NativeVarInfo[] ranges, Int32 rangeCount);
+
+        public void EmitDebugVar(DebugVarInfo debugVar)
+        {
+            int rangeCount = debugVar.Ranges.Count;
+            EmitDebugVar(_nativeObjectWriter, debugVar.Name, debugVar.TypeIndex, debugVar.IsParam, debugVar.Ranges.ToArray(), rangeCount);
```

Comment:
I think swapping rangeCount with debugVar.Ranges.ToArray() would be better. Normally we want to pass size of array followed by array.